xcbgen-rs/src/defs.rs

```diff
@@ -603,13 +1012,27 @@ impl NamedTypeRef {
     }
 }
 
+/// An alignment specification.
+///
+/// This structure represents a requirement that some byte position `pos` satisfies
+/// `(pos + offset) % align == 0`.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub struct Alignment {
     pub align: u32,
     pub offset: u32,
 }
 
 impl Alignment {
+    /// Advance this alignment specification by some variably sized object.
+    ///
+    /// The resulting value describes the alignment at the end of the variably sized object if it
+    /// is aligned by `self` at its beginning.
+    // FIXME: I am not sure this algorithm is correct (or that the alignment can even be specified
+    // in general). For example, imagine a list with elements of size 12 that is aligned at an 8
+    // byte boundary. The result of `Alignment { align: 8, offset: 0
+    // }.advance_variable_size(VariableSize { base: 0, incr: 12 })` has `.align == 8`, but only an
+    // alignment of 4 is actually guaranteed.
+    // I haven't thought about the `offset` calculation here.
```

Comment:
`incr` must be zero or a power of 2, so you would have instead `VariableSize { base: 0, incr: 4 }`, and the result of `advance_variable_size` would have `.offset == 4`.